src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -432,4 +443,26 @@ private function loadFromExtensions($content)
             $this->container->loadFromExtension($namespace, $values);
         }
     }
+
+    /**
+     * Checks the keywords used to define a service.
+     *
+     * @param string $id         The service name
+     * @param array  $definition The service definition to check
+     *
+     * @throws InvalidArgumentException When an invalid keyword is used
+     */
+    private static function checkDefinition($id, array $definition)
+    {
+        foreach ($definition as $key => $value) {
+            if (!in_array($key, static::$keywords)) {
+                throw new InvalidArgumentException(sprintf(
```

Comment:
This is a BC break and IMO you should just trigger a deprecation as until now we could use the newest keywords not yet implemented in older versions to ensure symfony 3.0 compatibility:
```yaml
services:
    foo:
        factory: [ '@bar', getFoo ]
        factory_service: bar
        factory_method: getFoo
```